extmod/vfs_posix_file.c

```diff
@@ -222,6 +226,9 @@ STATIC mp_uint_t vfs_posix_file_ioctl(mp_obj_t o_in, mp_uint_t request, uintptr_
 
 STATIC const mp_rom_map_elem_t vfs_posix_rawfile_locals_dict_table[] = {
     { MP_ROM_QSTR(MP_QSTR_fileno), MP_ROM_PTR(&vfs_posix_file_fileno_obj) },
+    #if MICROPY_PY_SYS_STDIO_BUFFER
+    { MP_ROM_QSTR(MP_QSTR_buffer), MP_ROM_PTR(&mp_stdio_buffer_obj) },
```

Comment:
Isn't this going to add `buffer` to all file objects, not just `stdin` and `stdout`?

If we do that, then it should provide a buffer object for this specific file instance rather than all objects returning a buffer for stdin/stdout.

Reply:
I intended this to just be for stdin and stdout, whereas stdin is read only and stdout is write only.  Similarly, stderr should also be write only.  Right now, it is not doing this, so I need to amend.

More broadly, maybe it's a better solution to have buffer on all file objects.  I will think of this as well.

Regarding code coverage, happy to also do this as well, naive question... but where are they?

Thank you very much for taking a look and providing feedback!

Reply:
Also fixed the nanbox check and incorporated using FileIO, will work on suggestions....

Reply:
File IO tests are in this directory: https://github.com/micropython/micropython/tree/master/tests/io

If `buffer` is conditionally compiled it probably makes sense to create a new file.

Reply:
Hope you are well, have an update on this one:

- I decided to go with adding the `buffer` attribute just on stdin/stdout/stderr files, keeping the scope as was intended for the PR.  For regular files, byte access is already there.

- As `MICROPY_PY_SYS_STDIO_BUFFER` already exists, seems natural to work with it, If `MICROPY_PY_SYS_STDIO_BUFFER` undefined, current bufferless stdio behavior is retained.

For tests, I did two items:

- I updated `file_stdio.py` to include stderr.  I thought it was missing, maybe there's a reason it shouldn't be there?

- I created a new test `file_stdio_buffer.py`.  This one checks to see if buffer attribute is available for stdio items and if so, run the test like `file_stdio.py`, if not, skip.

I wasn't sure if the test should be put in unix folder or io.  For different ports, the stdio tests would not work the same, for example, on esp32, `sys.stdin.fileno()` returns an error as the  `sys_stdio_mphal.c` implementation does not use files, but instead reads from `mp_hal_stdin_rx_chr` and writes to `mp_hal_stdout_tx_strn`.